File: Builder.php
<?php

interface UserBuilderInterface
{
  function setUserName($name);
  function setUserAge($age);
}

class User
{
  private $name;
  private $age;

  public function setName($name)
  {
    $this->name = $name;
  }

  public function getName()
  {
    return $this->name;
  }

  public function setAge($age)
  {
    $this->age = $age;
  }
  public function getAge()
  {
    return $this->age;
  }
}

class UserBuilder implements UserBuilderInterface
{
    private $user;

    public function __construct()
    {
      $this->user = new User();
    }

    public function setUserName($name)
    {
      $this->user->setName($name);
    }

    public function setUserAge($age)
    {
      $this->user->setAge($age);
    }

    public function getUser()
    {
      return $this->user;
    }
}

$builder = new UserBuilder();
$builder->setUserName("John Cena");
$builder->setUserAge(22);

var_dump($builder->getUser());

File: Builder.cs
<?php

interface IUserBuilderInterface
{
  void SetUserName(string name);
  void SetUserAge(int age);
}

class User
{
  private string _name;
  private int _age;

  public void SetName(string name)
  {
    _name = name;
  }

  public string GetName()
  {
    return _name;
  }

  public void SetAge(age)
  {
    _age = age;
  }
  public int GetAge()
  {
    return _age;
  }
}

class UserBuilder : IUserBuilderInterface
{
    private User _user;

    public UserBuilder()
    {
      _user = new User();
    }

    public void SetUserName(string name)
    {
      _user->SetName(name);
    }

    public void SetUserAge(int age)
    {
      _user->SetAge(age);
    }

    public string GetUser()
    {
      return _user;
    }
}

var builder = new UserBuilder();
builder.SetUserName("John Cena");
builder.SetUserAge(22);
